src/title.js

```diff
@@ -0,0 +1,437 @@
+
+		// Take care about proper `title` element structure.
+		model.document.registerPostFixer( writer => this._fixTitleContent( writer ) );
+
+		// Create and take care about proper position of a `title` element.
+		model.document.registerPostFixer( writer => this._fixTitleElement( writer ) );
+
+		// Prevent from adding extra paragraph after paste or enter.
+		model.document.registerPostFixer( writer => this._preventExtraParagraphing( writer ) );
+
+		// Attach `Title` and `Body` placeholders to the empty title and/or content.
+		this._attachPlaceholders();
+
+		// Attach Tab handling.
+		this._attachTabPressHandling();
+	}
+
+	/**
+	 * Sets the title of the document. This methods does not change any content outside the title element.
+	 *
+	 * @param {String} data Data to be set as a document title.
+	 */
+	setTitle( data ) {
+		const editor = this.editor;
+		const titleContent = editor.model.document.getRoot().getChild( 0 ).getChild( 0 );
+
+		editor.model.insertContent( editor.data.parse( data, 'title-content' ), titleContent, 'in' );
+	}
+
+	/**
+	 * Returns the title of the document. Note, that because this plugin does not allow any formatting inside
+	 * the title element, the output of this method will be a plain text, with no HTML tags. However, it
+	 * may contain some markers, like comments or suggestions. In such case, a special tag for the
+	 * marker will be included in the title text.
+	 *
+	 * @returns {String} Title of the document.
+	 */
+	getTitle() {
+		const title = this.editor.model.document.getRoot().getChild( 0 ).getChild( 0 );
+
+		return this.editor.data.stringify( title );
+	}
+
+	/**
+	 * Sets the body of the document.
+	 *
+	 * @returns {String} data Data to be set as a body of the document.
+	 */
+	setBody( data ) {
+		const editor = this.editor;
+		const root = editor.model.document.getRoot();
+		const range = editor.model.createRange(
+			editor.model.createPositionAt( root.getChild( 0 ), 'after' ),
+			editor.model.createPositionAt( root, 'end' )
+		);
+
+		editor.model.insertContent( editor.data.parse( data ), range );
+	}
+
+	/**
+	 * Returns the body of the document.
+	 *
+	 * @returns {String} Body of the document.
+	 */
+	getBody() {
+		const root = this.editor.model.document.getRoot();
+		const viewWriter = new ViewWriter();
+
+		// model -> view
+		const viewDocumentFragment = this.editor.data.toView( root );
+
+		// Remove title.
+		viewWriter.remove( viewWriter.createRangeOn( viewDocumentFragment.getChild( 0 ) ) );
+
+		// view -> data
+		return this.editor.data.processor.toData( viewDocumentFragment );
+	}
+
+	/**
+	 * Returns `title` element when is in the document. Returns `null` otherwise.
+	 *
+	 * @private
+	 * @returns {module:engine/model/element~Element|null}
+	 */
+	_getTitleElement() {
+		const root = this.editor.model.document.getRoot();
+
+		return Array.from( root.getChildren() ).find( element => element.is( 'title' ) );
```

Comment:
This will return `undefined` not `null` if element is not found in the array.

Reply:
Good catch.